src/api-client.ts

```diff
@@ -43,48 +43,65 @@ export class ApiClient {
   }
 
   async getJobDetails(): Promise<JobDetails> {
-    const detailsURL = `/update_jobs/${this.params.jobId}/details`
+    const detailsURL = `${this.params.dependabotApiUrl}/update_jobs/${this.params.jobId}/details`
     try {
-      const res: any = await this.client.get(detailsURL, {
-        headers: {Authorization: this.params.jobToken}
-      })
-      if (res.status !== 200) {
+      const res = await this.getJsonWithRetry<any>(
+        detailsURL,
+        this.params.jobToken
+      )
+      if (res.statusCode !== 200) {
         throw new JobDetailsFetchingError(
-          `fetching job details: unexpected status code: ${res.status}`
+          `fetching job details: unexpected status code: ${
+            res.statusCode
+          }: ${JSON.stringify(res.result)}`
+        )
+      }
+      if (res.result == null) {
+        throw new JobDetailsFetchingError(
+          `fetching job details: missing response`
         )
       }
 
-      return res.data.data.attributes
+      return res.result.data.attributes
```

Comment:
can any of these be `falsy` and throw a null ref?

Reply:
It wouldn't be expected so a null ref failing the job would be OK.